tests/test_signals/test_ebsd_master_pattern.py

```diff
@@ -758,3 +775,487 @@ def test_vector2xy(self):
+        r_ang, n_ang = self.get_r_ang_and_n_ang(cds_px_int, det, d_vec_n, dddd)
+
+        return a_ang, d_ang, r_ang, n_ang
+
+    @pytest.mark.parametrize(
+        "tilt, azimuthal, euler_2, zone_axes",
+        [
+            (0.0, 0.0, 0.0, [[1, 0, 1], [0, 0, 1], [1, 1, 2]]),
+            (0.0, 0.0, 1.2, [[1, 0, 1], [0, 0, 1], [1, 1, 2]]),
+            (40.0, 0.0, 0.0, [[1, 0, 1], [1, 0, 0], [1, -2, 1]]),
+            (40.0, 0.0, 1.2, [[1, 0, 1], [1, 0, 0], [1, -2, 1]]),
+            (0.0, 40.0, 0.0, [[1, 0, 1], [1, 1, 0], [1, 2, 1]]),
+            (0.0, 40.0, 1.2, [[1, 0, 1], [1, 1, 0], [1, 2, 1]]),
+            (40.0, 40.0, 0.0, [[1, 0, 1], [1, 0, 0], [3, 1, 0]]),
+            (40.0, 40.0, 1.2, [[1, 0, 1], [1, 0, 0], [3, 1, 0]]),
+        ],
+    )
+    def test_fit_detector_orientation(self, tilt, azimuthal, euler_2, zone_axes):
+        """
+        Check that the EBSD pattern simulated from a master
+        pattern and the associated
+        GeometricalKikuchiPatternSimulation match perfectly,
+        for various detector orientations.
+
+        4 sets of angles are returned by self.calculate_fit().
+        See the doctstring of that function for details.
+
+        Here we assert that the first set of angles are all
+        zero, that the second and third sets are equal, and
+        that the fourth set are all zero. If these conditions
+        are all met, the GeometricalKikuchiPatternSimulation
+        should match the EBSD pattern simulated from a
+        master pattern perfectly for the given detector
+        orientations.
+
+
+        Parameters
+        ----------
+        tilt : Float
+            The detector tilt angle in degrees (i.e. detector.tilt).
+            Detector Euler angle PHI (EBSDDetector.euler[1]) == 90 + tilt
+        azimuthal : Float
+            The detector azimuthal angle in degrees (i.e. detector.azimuthal).
+            Detector Euler angle phi1 (EBSDDetector.euler[0]) == azimuthal
+        euler_2 : Float
+            The detector Euler angle phi2 (EBSDDetector.euler[2]) in deg.
+        zone_axes : List or np.ndarray
+            List/array containing three lists, each containing a set
+            of uvw indices describing a zone axis on the pattern,
+            e.g. [[0,0,1], [1,1,0], [1,2,3]].
+
+        Returns
+        -------
+        None.
+
+        """
+        angles = self.calculate_fit(tilt, azimuthal, euler_2, zone_axes)
+
+        assert np.allclose(angles[0], 0.0)
+        assert np.allclose(angles[1], angles[2])
+        assert np.allclose(angles[3], 0.0)
+
+
+def index_row_in_array(myarray, myrow):
+    """Check if the row "myrow" is present in the array "myarray".
+    If it is, return an int containing the row index of the first
+    occurrence. If the row is not present, return None.
+    """
+    loc = np.where((myarray == myrow).all(-1))[0]
+    if len(loc) > 0:
+        return loc[0]
+    return None
+
+
+def get_coordinate_conversions(detector):
```

Comment:
It looks like the next two functions may be useful to have as functionality accessible via the EBSD detector?

Reply:
Yes, they could easily be moved into the EBSDDetector class.